precompiles/staking/tx_test.go

```diff
@@ -918,45 +937,47 @@ func (s *PrecompileTestSuite) TestCancelUnbondingDelegation() {
 	for _, tc := range testCases {
 		s.Run(tc.name, func() {
 			s.SetupTest()
+			ctx = s.network.GetContext()
+			stDB := s.network.GetStateDB()
 
-			var contract *vm.Contract
-			contract, s.ctx = testutil.NewPrecompileContract(s.T(), s.ctx, s.address, s.precompile, tc.gas)
+			contract, ctx := testutil.NewPrecompileContract(s.T(), ctx, s.keyring.GetAddr(0), s.precompile, tc.gas)
 
 			if tc.expError {
-				bz, err := s.precompile.CancelUnbondingDelegation(s.ctx, s.address, contract, s.stateDB, &method, tc.malleate(s.validators[0].OperatorAddress))
+				bz, err := s.precompile.CancelUnbondingDelegation(ctx, s.keyring.GetAddr(0), contract, stDB, &method, tc.malleate(s.network.GetValidators()[0].OperatorAddress))
 				s.Require().ErrorContains(err, tc.errContains)
 				s.Require().Empty(bz)
 			} else {
 				undelegateArgs := []interface{}{
-					s.address,
-					s.validators[0].OperatorAddress,
+					s.keyring.GetAddr(0),
+					s.network.GetValidators()[0].OperatorAddress,
 					big.NewInt(1000000000000000000),
 				}
 
-				err := s.CreateAuthorization(s.address, staking.UndelegateAuthz, nil)
+				err := s.CreateAuthorization(ctx, s.keyring.GetAddr(0), staking.UndelegateAuthz, nil)
 				s.Require().NoError(err)
 
-				_, err = s.precompile.Undelegate(s.ctx, s.address, contract, s.stateDB, &undelegateMethod, undelegateArgs)
+				_, err = s.precompile.Undelegate(ctx, s.keyring.GetAddr(0), contract, stDB, &undelegateMethod, undelegateArgs)
 				s.Require().NoError(err)
 
-				_, err = s.app.StakingKeeper.GetDelegation(s.ctx, s.address.Bytes(), sdk.ValAddress(s.validators[0].GetOperator()))
+				valAddr, err := sdk.ValAddressFromBech32(s.network.GetValidators()[0].GetOperator())
+				s.Require().NoError(err)
+
+				_, err = s.network.App.StakingKeeper.GetDelegation(ctx, s.keyring.GetAccAddr(0), valAddr)
 				s.Require().Error(err)
-				s.Require().Contains("not found", err.Error())
+				s.Require().Contains("no delegation for (address, validator) tuple", err.Error())
 
-				err = s.CreateAuthorization(s.address, staking.CancelUnbondingDelegationAuthz, nil)
+				err = s.CreateAuthorization(ctx, s.keyring.GetAddr(0), staking.CancelUnbondingDelegationAuthz, nil)
 				s.Require().NoError(err)
 
-				bz, err := s.precompile.CancelUnbondingDelegation(s.ctx, s.address, contract, s.stateDB, &method, tc.malleate(s.validators[0].OperatorAddress))
+				bz, err := s.precompile.CancelUnbondingDelegation(ctx, s.keyring.GetAddr(0), contract, stDB, &method, tc.malleate(s.network.GetValidators()[0].OperatorAddress))
```

Comment:
the new testing suite gives us the change to really improve the readalibilty of the tests IMO


```suggestion
			ctx = s.network.GetContext()
			stDB := s.network.GetStateDB()
            delegator := s.keyring.GetKey(0)

			contract, ctx := testutil.NewPrecompileContract(s.T(), ctx, delegator.Addr, s.precompile, tc.gas)

			if tc.expError {
				bz, err := s.precompile.CancelUnbondingDelegation(ctx, delegator.Addr, contract, stDB, &method, tc.malleate(s.network.GetValidators()[0].OperatorAddress))
				s.Require().ErrorContains(err, tc.errContains)
				s.Require().Empty(bz)
			} else {
				undelegateArgs := []interface{}{
					delegator.Addr,
					s.network.GetValidators()[0].OperatorAddress,
					big.NewInt(1000000000000000000),
				}

				err := s.CreateAuthorization(ctx, delegator.Addr, staking.UndelegateAuthz, nil)
				s.Require().NoError(err)

				_, err = s.precompile.Undelegate(ctx, delegator.Addr, contract, stDB, &undelegateMethod, undelegateArgs)
				s.Require().NoError(err)

				valAddr, err := sdk.ValAddressFromBech32(s.network.GetValidators()[0].GetOperator())
				s.Require().NoError(err)

				_, err = s.network.App.StakingKeeper.GetDelegation(ctx, delegator.AccAddr, valAddr)
				s.Require().Error(err)
				s.Require().Contains("no delegation for (address, validator) tuple", err.Error())

				err = s.CreateAuthorization(ctx, delegator.Addr, staking.CancelUnbondingDelegationAuthz, nil)
				s.Require().NoError(err)

				bz, err := s.precompile.CancelUnbondingDelegation(ctx, delegator.Addr, contract, stDB, &method, tc.malleate(s.network.GetValidators()[0].OperatorAddress))
```